lib/login/login-controller.js

```diff
@@ -0,0 +1,46 @@
+var CONSTANTS = require('../constants');
+
+/**
+ *
+ * Controller for user login / logout
+ *
+ * @param $timeout
+ * @param {UserMediatorService} userMediatorService
+ *
+ * @constructor
+ */
+function LoginCtrl($timeout, userMediatorService) {
+  var self = this;
+  self.loginErrorMessage = "";
+  self.loginMessages = {success: false, error: false};
+
+  userMediatorService.userHasSession().then(function(hasSession) {
+    $timeout(function() {
+      self.hasSession = hasSession;
+    });
+  });
+
+  self.login = function(valid) {
+    if (valid) {
+      self.loginErrorMessage = "";
+      self.loginMessages.error = false;
+      userMediatorService.authenticateUser(self.username, self.password)
```

Comment:
@TommyJ1994 - marking as possible problem with password exposure (vulnerable module can mount here and backup passwords to hacker server.